tests/FSharp.Compiler.ComponentTests/Conformance/TypesAndTypeConstraints/IWSAM/IWSAMTests.fs

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+module FSharp.Compiler.ComponentTests.Conformance.TypeAndTypeConstraints.IWSAM
+
+open Xunit
+open System.IO
+open FSharp.Test
+open FSharp.Test.Compiler
+
+let typesModule =
+    FSharp (loadSourceFromFile (Path.Combine(__SOURCE_DIRECTORY__,  "Types.fs")))
+    |> withName "Types"
+    |> withLangVersionPreview
+    |> withOptions ["--nowarn:3535"]
+
+let setupCompilation compilation =
+    compilation
+    |> asExe
+    |> withLangVersionPreview
+    |> withReferences [typesModule]
+
+[<Theory; Directory(__SOURCE_DIRECTORY__ + "/testFiles")>]
+let ``IWSAM test files`` compilation =
+    compilation
+    |> setupCompilation
+    |> compileAndRun
+    |> shouldSucceed
+
+[<Theory>]
+[<InlineData("let inline f0 (x: ^T) = x",
+             "val inline f0: x: ^T -> ^T")>]
+[<InlineData("""
+             let inline f0 (x: ^T) = x
+             let g0 (x: 'T) = f0 x""",
+             "val g0: x: 'T -> 'T")>]
```

Comment:
This is failing, giving instead `obj -> obj`. Even though in FSI it works as expected.